addOns/exim/src/main/java/org/zaproxy/addon/exim/pcap/PcapImporter.java

```diff
@@ -19,33 +19,137 @@
  */
 package org.zaproxy.addon.exim.pcap;
 
+import io.pkts.streams.TcpStream;
 import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.history.ExtensionHistory;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.ui.ProgressPane;
 import org.zaproxy.addon.commonlib.ui.ProgressPaneListener;
+import org.zaproxy.zap.utils.ThreadUtils;
 
 public class PcapImporter {
 
-    private ProgressPaneListener progressListener;
+    private static final Logger LOGGER = LogManager.getLogger(PcapImporter.class);
+
+    private static ExtensionHistory extHistory;
+
+    private ProgressPane progressPane = null;
+    private ProgressPaneListener progressListener = null;
     private boolean success;
 
     public PcapImporter(File file) {
-        this(file, null);
+        importPcapFile(file);
+    }
+
+    public PcapImporter(File file, ProgressPane progressPane) {
```

Comment:
Better to change the `ProgressPaneListener` to allow to set the total tasks, rather than passing the pane.